packages/number-field/test/visual/lumo/number-field.test.js

```diff
@@ -28,6 +28,11 @@ describe('number-field', () => {
     await visualDiff(div, 'readonly');
   });
 
+  it('invalid', async () => {
+    element.invalid = true;
+    await visualDiff(div, 'invalid');
+  });
```

Comment:
When the field is invalid without an error message, it is not expected that it would have the `has-error-message` attribute, correct?

Just to be sure, does the test assert that case?

Reply:
1. We only set `has-error-message` when both `invalid` and `error-message` are set.
2. No, this particular test does not cover it, and is not supposed to do it.